docs/developer/releasing.md

```diff
@@ -0,0 +1,22 @@
+# Releasing
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared cluster-api controller manager
+2. A git tag for providers to use
+
+### Artifact locations
+
+1. The container image is found in the registry `gcr.io/k8s-cluster-api` with an
+   image name of `cluster-api-controller` and a tag that matches the release
+   version. For example, in the `0.0.0-alpha.4` release, the container image
+   location is `gcr.io/k8s-cluster-api/cluster-api-controller:0.0.0-alpha.4`
+
+# Process
+
+1. Push a git tag
+2. Build and push the container image with `make docker-push`. Note: This
```

Comment:
You either need to set CONTROLLER_IMG before running the make command or edit the Makefile so that the pushed image has the correct tag.

Reply:
If we have to do a PR dance to cut a release (to embed version numbers) then it might make sense to have that include editing the version in the Makefile too).